LayoutTests/imported/w3c/web-platform-tests/html/semantics/document-metadata/the-link-element/stylesheet-bad-mime-type.html

```diff
@@ -0,0 +1,49 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>stylesheets served with bad MIME types</title>
+<link rel="author" title="Michael[tm] Smith" href="mailto:mike@w3.org">
+<link rel="help" href="https://html.spec.whatwg.org/#link-type-stylesheet:process-the-linked-resource">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+  async_test((t) => {
+    const link = document.createElement("link");
+    let errorFired = false;
+    link.rel = "stylesheet";
+    link.href = "resources/stylesheet-bad-mime-type.css";
+    link.addEventListener("load", t.unreached_func());
+    document.head.append(link);
+    t.step_timeout(() => t.done(), 2000);
+  }, "'load' event does not fire at link@rel=stylesheet having non-empty resource with bad MIME type");
```

Comment:
Why not have error and load event listeners for all of these tests instead of a 2s timeout? For `link` elements one of those events should fire and it seems preferable to wait for that.

Reply:
> Why not have error and load event listeners for all of these tests instead of a 2s timeout? For `link` elements one of those events should fire and it seems preferable to wait for that.

I don’t understand enough about the test harness to make it wait for an event to be fired. Do I need to use `promise_test(…)`? Or else is there some existing test I could use as a model?

Reply:
You want to use `t.step_func_done` as a wrapper of the error event listener.

Reply:
https://github.com/web-platform-tests/wpt/blob/master/xhr/event-upload-progress-crossorigin.any.js has an example. You could add further asserts in a callback to `step_func_done` (it's equivalent to `step_func` but it calls `done` in the end) if you want.

Reply:
Thanks much — updated and pushed just now